Clean Up Build

hazen opened the pull request:
- Removed the Scons files
- Updated build instructions for multiple platforms
- Build PBC .h and .c files from riak_pb instead of static versions
- Updated build version to 1.0
- Addresses https://github.com/basho/riak-c-client/issues/34, https://github.com/basho/riak-c-client/issues/35 and https://github.com/basho/riak-c-client/issues/36

Commit:
Update Build and Instructions
- Remove Scons files
- Update README for multiple platforms
- Generate PBC headers from riak_pb
- Update version to 1.0

bookshelfdave commented on README.md:
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
-	cd riak-c-client
-	./autogen.sh
-	make
+Be sure Xcode, command-line tools and [Home Brew](http://brew.sh/) installed first
```
add a period at the end

bookshelfdave commented on README.md:
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
+* autoconf
+* libtool
 * libevent
 * protobuf
```
is it worth pointing out v2.5.0 here?

hazen replied:
Yes

bookshelfdave commented on README.md:
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
+* libtool
 * libevent
 * protobuf
 * protobuf-c
```
same with protobuf-c-0.15

hazen replied:
Yes.   Same with libevent-2.0.21

bookshelfdave commented:
does this PR bring RCC to v1.0? I see it stamped in `Makefile.am`.
If so, we'll need to tag/release etc.

hazen commented:
Upped version to 1.0.  Agree we should tag when it's "done."

Commit:
Add README placeholder for proto files

Commit:
Simplify bulid by putting riak_pb under riak-c-client

Commit:
Add `make install` step to build instructions

Commit:
Merge branch 'develop' into bugfix/bch/cleanup-build

bookshelfdave commented:
This builds fine if I follow the readme :-) 

I think we need to:
a) give the user a better error message if they haven't cloned riak_pb into this src tree
b) add a script + metadata file including riak_pb vsn# to automate the download
   (similar to rebar deps/locked deps)

bookshelfdave commented:
I'd ask clients@basho about the version number. We probably want to avoid 1.0 in the configs until this is actually released.

Commit:
Change build per @metadave's comments
- Set version to 0.5 for safety
- Extended autogen.sh to pull riak_pb version automagically
- Added support for FreeBSD

bookshelfdave commented:
@javajolt I pushed some readme updates directly to your branch. Can you give them a quick review?

bookshelfdave commented:
:+1:

hazen commented:
@metadave I don't see any changes to the README file on the `bugfix/bch/cleanup-build` branch. Did you push?

Commit:
added riak_pb note to README.md

bookshelfdave commented:
derp, pushed.
